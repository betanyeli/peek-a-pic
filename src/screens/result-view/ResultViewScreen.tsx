import { Image, Dimensions, View } from 'react-native';
import React, { useState } from 'react';
import styles from './ResultViewScreen.styles';
import { SafeAreaView } from 'react-native-safe-area-context';
import ErrorAnimation from './ErrorAnimation';
import { ActivityIndicator, Button } from 'react-native-paper';
import SharedSocialMediaComponent from '../../components/shared-social-media/SharedSocialMediaComponent';
import FooterComponent from '../../components/footer/FooterComponent';

const ResultViewScreen = ({ route, navigation }: any) => {
  const { response } = route.params;
  const widthImg = Dimensions.get('window').width;
  const heightImg = 250;
  const labelButton: string = response?.errorCode ? 'Try again' : 'Go back';
  const defaultImg: string =
    'https://images.unsplash.com/photo-1618826411640-d6df44dd3f7a?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2048&q=80';

  const [loading, setIsLoading] = useState<boolean>(false);

  return (
    <SafeAreaView style={styles.container}>
      {loading && <ActivityIndicator animating />}
      {response?.errorCode && <ErrorAnimation />}
      {response?.assets &&
        response?.assets.map(
          ({ uri, width }: { uri: string; width: number }) => (
            <View key={uri} style={styles.response}>
              <Image
                resizeMode="contain"
                resizeMethod="scale"
                style={{
                  width: widthImg - 48,
                  height: heightImg,
                }}
                source={{ uri: width > 0 ? uri : defaultImg }}
                onLoadStart={() => setIsLoading(true)}
                onLoadEnd={() => setIsLoading(false)}
              />
              <SharedSocialMediaComponent />
            </View>
          ),
        )}

      <FooterComponent showLogo={false}>
        <Button
          mode="contained"
          icon="rotate-left"
          onPress={() => {
            navigation.goBack();
          }}>
          {labelButton}
        </Button>
      </FooterComponent>
    </SafeAreaView>
  );
};

export default ResultViewScreen;
