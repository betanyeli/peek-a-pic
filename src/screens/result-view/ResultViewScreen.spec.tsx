import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import ResultViewScreen from './ResultViewScreen';

describe('ResultViewScreen test', () => {
  it('Should match with snapshot', () => {
    const mockedParams = {
      route: { params: { response: {} } },
      navigation: '',
    };
    const { toJSON } = render(<ResultViewScreen {...mockedParams} />);
    waitFor(() => expect(toJSON()).toMatchSnapshot());
  });
});
