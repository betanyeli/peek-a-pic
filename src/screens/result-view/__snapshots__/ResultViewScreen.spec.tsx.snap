// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultViewScreen test Should match with snapshot 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    {
      "backgroundColor": "#000",
      "flex": 1,
      "padding": 40,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 10,
        "marginHorizontal": 40,
        "marginTop": 30,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignSelf": undefined,
          "bottom": undefined,
          "end": undefined,
          "left": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(103, 80, 164, 1)",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "marginLeft": 12,
                      "marginRight": -4,
                    },
                    {
                      "marginLeft": 16,
                      "marginRight": -16,
                    },
                    false,
                  ]
                }
                testID="button-icon-container"
              >
                <Text
                  accessibilityElementsHidden={true}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  â–¡
                </Text>
              </View>
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Go back
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
